src/structured_config.py

```diff
@@ -57,6 +57,7 @@ class CharmConfig(BaseConfigModel):
     log_cleanup_policy: str
     log_message_timestamp_type: str
     ssl_cipher_suites: Optional[str]
+    ssl_principal_mapping_rules: Optional[str]
```

Comment:
**todo:** I would add a few validations here, as it's highly likely someone will mess up the regex and then get confused. 

Off the top of my head, maybe tests for:

```
- Is the config a list of strings? 
- Is each item in the list of strings valid regex?
    - try: ; re.compile(pattern); except re.error: ; raise 
```

Reply:
I think using `re` lib for this is a great idea! I'll add that.

Reply:
Note from debugging, you'll have to only regex up to the first `/`, else the regex won't compile.

Reply:
It should also not be optional, as we have a default in the `config.yaml`